portfolio/src/main/webapp/commentsection.js

```diff
@@ -13,27 +13,69 @@
 // limitations under the License.
 
 function getComments() {
-    const showNum = getCommentLimit();
-    document.getElementById('shownum').value = parseInt(showNum);
-    fetch('/comment-data?shownum='+showNum).then(response => response.json()).then((comments) => {
+    const perBlock = getCommentsPerBlock();
+    document.getElementById('perblock').value = parseInt(perBlock);
+    fetch('/comment-data?perblock='+(perBlock*blockIdx)).then(response => response.json()).then((data) => {
+        let comments = data.comments;
         const commentsContainer = document.getElementById("commentsection-container");
         commentsContainer.innerHTML = "";
         comments.forEach((line) => {
             commentsContainer.appendChild(createComment(line));
         });
+        if(data.totalCommentCount <= perBlock * blockIdx) {
+            let showmorebutton = document.getElementById("showmorebutton");
+            showmorebutton.style.display = "none";
+        }
     });
 }
 
+var blockIdx = 1;
+
+function showMore() {
+    blockIdx++;
+    getComments();
+}
+
 function createComment(commentData) {
     let res = document.createElement("div");
     res.setAttribute("class", "single-comment");
     
-    let username = document.createElement("div");
+    let voting = document.createElement("div");
+    voting.setAttribute("class", "voting");
+
+    let upvote = document.createElement("div");
+    upvote.setAttribute("class", "upvote");
+    let upvoteSymbol = document.createElement("div");
+    upvoteSymbol.setAttribute("class", "upvoteSymbol");
+    upvoteSymbol.innerText = "▲";
+    let upvoteCount = document.createElement("div");
+    upvoteCount.setAttribute("class", "upvoteCount");
+    upvoteCount.innerText = "0";
+    upvote.appendChild(upvoteSymbol);
+    upvote.appendChild(upvoteCount);
+
+    let downvote = document.createElement("div");
+    downvote.setAttribute("class", "downvote");
+    let downvoteSymbol = document.createElement("div");
+    downvoteSymbol.setAttribute("class", "downvoteSymbol");
+    downvoteSymbol.innerText = "▼";
+    let downvoteCount = document.createElement("div");
+    downvoteCount.setAttribute("class", "downvoteCount");
+    downvoteCount.innerText = "0";
+    downvote.appendChild(downvoteSymbol);
+    downvote.appendChild(downvoteCount);
+
+    voting.appendChild(upvote);
+    voting.appendChild(downvote);
+
+    let commentHeader = document.createElement("div");
+    commentHeader.setAttribute("class", "comment-header");
+
+    let username = document.createElement("span");
     username.setAttribute("class", "username");
-    username.innerText = commentData.username + ":";
+    username.innerText = commentData.username;
 
-    let time = document.createElement("div");
-    time.setAttribute("class", "time");
+    let timeText = "";
     let val = -1;
     let units = "";
     if(commentData.time < 1000*60) {
```

Comment:
Here you use a lot of numbers. Although it almost does the right thing, it easily causes some errors. Actually there is a small error in the code. So the following code is more error-proof.

let secondInMs = 1000;
let minuteInMs = 60 * secondInMs;
let hourInMs = 60 * minuteInMs;
let dayInMs = 24 * hourInMs;
let yearInMs = 365 * dayInMs;
if(commentData.time < minuteInMs) {
    val = Math.floor(commentData.time/secondInMs);
    units = "second";
} else if(commentData.time < hourInMs) {
    val = Math.floor(commentData.time/minuteInMs);
    units = "minute";
} else if(commentData.time < dayInMs) {
    val = Math.floor(commentData.time/hourInMs);
    units = "hour";
} else if(commentData.time < yearInMs) {
    val = Math.floor(commentData.time/dayInMs);
    units = "day";
} else {
    val = Math.floor(commentData.time/yearInMs);
    units = "year";
}

Reply:
Thanks! I've cleaned up this part of the code according to your suggestion.